tests/filter.json

```diff
@@ -829,6 +845,22 @@
         {"a": 0.01, "d": "e"}
       ]
     },
+    {
+      "name": "equals number, exponent leading 0",
+      "selector" : "$[?@.a==1e-02]",
+      "document" : [{"a": 0.01, "d": "e"}, {"a":0.02, "d": "f"}, {"a":"0.01", "d": "g"}],
+      "result": [
+        {"a": 0.01, "d": "e"}
+      ]
+    },
+    {
+      "name": "equals number, exponent 00",
+      "selector" : "$[?@.a==1e+00]",
```

Comment:
Do we have a single-zero exponent test?  (1e0, 1e+0, 1e+00)

Or negative zero exponent? (1e-0)

Reply:
I don't think there were any existing tests; I have added some now.